src/v/archival/archiver_service.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "archival/ntp_archiver_service.h"
+#include "archival/probe.h"
+#include "archival/types.h"
+#include "cluster/fwd.h"
+#include "cluster/partition_leaders_table.h"
+#include "container/fragmented_vector.h"
+
+#include <seastar/core/shared_ptr.hh>
+
+namespace archival {
+
+/// Maintain an archiver instance per partition. Get notifications
+/// on membership/leadership changes and make corresponding changes
+/// to the set of managed instances.
+class archiver_service {
+    friend struct managed_partition;
+    friend struct managed_partition_stm;
+    class impl;
+
+public:
+    archiver_service(
```

Comment:
I'd wish we had a better/different name for this. We already have the ntp_archiver_service. ntp_archiver_manager?